src/components/input-panel/spec-editor/renderer.js

```diff
@@ -86,6 +79,23 @@ export default class Editor extends React.Component {
       return null;
     }
   }
+
+  updateSpec(spec) {
+    let schema, parsedMode;
+    try {
+      schema = JSON.parse(spec).$schema;
+    } catch (err) {
+      console.warn('Error parsing json string');
+    } 
+    if (schema) {
+      parsedMode = parser(schema).library;
+    }  
+    if (parsedMode === MODES.Vega || (!parsedMode && this.props.mode === MODES.Vega)) {
```

Comment:
@domoritz but then we need to have more lines to handle the case when schema is not defined